### ISSUE TITLE
Setting larger arrow size causes text to be clipped

### ISSUE THREAD

**Issue by dbrant** [skydoves/Balloon#89]
When we increase the size of the arrow, it seems to be causing the entire balloon to shrink, which also causes the text to be clipped:

![image](https://user-images.githubusercontent.com/1682214/90149662-bf042c80-dd52-11ea-8d62-8e75e199e9eb.png)

Our settings are roughly like this:

```
Balloon.Builder(context)
    .setText(text)
    .setArrowConstraints(ArrowConstraints.ALIGN_ANCHOR)
    .setArrowOrientation(ArrowOrientation.TOP)
    .setArrowSize(24)
    .setPadding(16)
    .setTextSize(14f)
    .build();
```

**Comment by skydoves** [comment]
Hi, @dbrant,

I just fixed and published a new version `1.2.2-alpha01` for testing this issue.
Could you please test aging with this version?

Thank you for your issue!

**Comment by dbrant** [comment]
Thanks, that looks much better.
But now, the tooltip no longer has any margins on the sides, and calling `setSpace()` doesn't seem to have any effect.

![image](https://user-images.githubusercontent.com/1682214/90175330-419de380-dd75-11ea-8ee5-28354e6d3d9e.png)

**Comment by skydoves** [comment]
The `setSpace()` method only works with `setWidthRatio(1.0f)`.
Because it gives spaces from the display sides.
In this case, It seems to need `xOff` attributes for giving spaces from the left side.

**Comment by dbrant** [comment]
I may still be doing something wrong, but providing `xOff` seems to have no effect, and neither does calling `setSpace` with `setWidthRatio(1.0f)`, the balloon still has no left margin.

**Comment by skydoves** [comment]
How about this way?
```kotlin
this.customListBalloon.showAlignBottom(it, (balloon.getMeasureWidth() / 2) - (targetView.width / 2) + (desired margin), 0)
```
The `PopupWindow`'s x position is lower than our expected. (actually not 0 in that case)

**Comment by dbrant** [comment]
This still does not have any effect on the margins (using `1.2.2-alpha01`):

![image](https://user-images.githubusercontent.com/1682214/90430222-12e57d00-e095-11ea-8851-aca588c4e2d4.png)

Here is the same code using the latest stable `1.2.1` (setting `xOff` does not have any effect either, but it looks like there are some default margins):

![image](https://user-images.githubusercontent.com/1682214/90430725-f269f280-e095-11ea-9420-3aea40904175.png)

**Comment by skydoves** [comment]
Hmm, how about giving huge xOff enough like 800 or 900 positive integer value? (in `1.2.2-alpha01`)
I will debug calculating the exact position of the popup.

**Comment by skydoves** [comment]
Hi, @dbrant.
It is released a new version `1.2.2`.
Now we can use below new methods without any constraints.

### Margin
If the location of the balloon according to the anchor is located at the boundaries on the screen,
the balloon will be stick to the end of the screen. In that case, we can resolve by giving margins to the balloon.

```kotlin
.setMargin(12) // sets the margin on the balloon all directions.
.setMarginLeft(14) // sets the left margin on the balloon.
.setMarginRight(14) // sets the right margin on the balloon.
```

Thanks for your issue!
